File: scripts/test_grok.py
import os
import requests
import json
from dotenv import load_dotenv

# Cargar variables desde el archivo .env en la raíz del proyecto
load_dotenv(os.path.join(os.path.dirname(__file__), '..', '.env'))

api_key = os.getenv('XAI_API_KEY')

print("--- goalworld Grok (xAI) Test Script ---")

if not api_key:
    print("❌ Error: Falta XAI_API_KEY en el archivo .env")
    exit(1)

url = "https://api.x.ai/v1/chat/completions"

headers = {
    "Content-Type": "application/json",
    "Authorization": f"Bearer {api_key}"
}

data = {
    "model": os.environ.get("XAI_MODEL", "grok-4.3"),
    "messages": [
        {"role": "system", "content": "Eres el motor de inteligencia artificial de goalworld. Hablas de forma épica, futbolera y con terminología Web3/Solana."},
        {"role": "user", "content": "Saluda a la comunidad de goalworld y confirma que tu motor de IA está en línea."}
    ],
    "stream": False,
    "temperature": 0.7
}

try:
    print("Conectando con SuperGrok...")
    response = requests.post(url, headers=headers, data=json.dumps(data))
    
    if response.status_code == 200:
        result = response.json()
        content = result['choices'][0]['message']['content']
        print(f"\n🤖 Grok dice:\n{content}\n")
        print("✅ ¡Conexión exitosa!")
    else:
        print(f"❌ Error en la API ({response.status_code}): {response.text}")

except Exception as e:
    print(f"❌ Error inesperado: {e}")

File: api/src/scripts/test_grok.ts
import * as dotenv from 'dotenv';
import * as path from 'path';

// Cargar .env desde la raíz del proyecto
dotenv.config({ path: path.resolve(__dirname, '../../../.env') });

const XAI_API_KEY = process.env.XAI_API_KEY;

async function testGrok() {
  console.log('--- goalworld Grok (xAI) Node.js Test ---');

  if (!XAI_API_KEY) {
    console.error('❌ Error: Falta XAI_API_KEY en el archivo .env');
    return;
  }

  try {
    console.log('Conectando con SuperGrok...');
    
    const response = await fetch('https://api.x.ai/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${XAI_API_KEY}`
      },
      body: JSON.stringify({
        model: process.env.XAI_MODEL || 'grok-4.3',
        messages: [
          { role: 'system', content: 'Eres el motor de inteligencia artificial de goalworld. Hablas de forma épica, futbolera y con terminología Web3/Solana.' },
          { role: 'user', content: 'Saluda a la comunidad de goalworld y confirma que tu motor de IA está en línea.' }
        ],
        temperature: 0.7
      })
    });

    if (response.ok) {
      const result = await response.json() as any;
      const content = result.choices[0].message.content;
      console.log(`\n🤖 Grok dice:\n${content}\n`);
      console.log('✅ ¡Conexión exitosa!');
    } else {
      const errorText = await response.text();
      console.error(`❌ Error en la API (${response.status}):`, errorText);
    }
  } catch (error) {
    console.error('❌ Error inesperado:', error);
  }
}

testGrok();
